src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,226 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_DARUMA_VANISH);
+        BgIkanaDharma_BeginWaitToStartCutscene(this);
+    } else if ((this->dyna.actor.flags & ACTOR_FLAG_40) == ACTOR_FLAG_40 && sFirstHitBgIkanaDharma == NULL &&
+               this->dyna.actor.xzDistToPlayer < 420.0f) {
+        tempAngle1 = BINANG_SUB(this->dyna.actor.yawTowardsPlayer, player->actor.shape.rot.y);
+        tempAngle1 = ABS_ALT(tempAngle1);
+
+        if (tempAngle1 > 0x4000) {
+            Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
+
+void BgIkanaDharma_BeginWaitToStartCutscene(BgIkanaDharma* this) {
+    ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    this->actionFunc = BgIkanaDharma_WaitToStartCutscene;
+}
+
+void BgIkanaDharma_WaitToStartCutscene(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        BgIkanaDharma_BeginCutsceneState(this);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void BgIkanaDharma_BeginCutsceneState(BgIkanaDharma* this) {
+    this->cutsceneFramesRemaining = 0xA;
+    this->actionFunc = BgIkanaDharma_UpdateCutscene;
+}
+
+void BgIkanaDharma_UpdateCutscene(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    if (this->cutsceneFramesRemaining > 0) {
+        this->cutsceneFramesRemaining--;
+
+        if (this->cutsceneFramesRemaining == 0) {
+            if (sFirstHitBgIkanaDharma == this) {
+                sFirstHitBgIkanaDharma = NULL;
+            }
+
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+
+    if (Math_StepToF(&this->dyna.actor.scale.y, 0.0f, 1.0f / 300.0f) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        this->dyna.actor.scale.x = this->dyna.actor.scale.y * 3.0f;
+        this->dyna.actor.scale.z = this->dyna.actor.scale.y * 3.0f;
+        BgIkanaDharma_CreateParticles(this, globalCtx);
+    }
+}
+
+void BgIkanaDharma_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc == BgIkanaDharma_UpdateNormalState) {
+        DynaPolyActor* actorBelow;
+        s32 pad[2];
+
+        func_800B4AEC(globalCtx, &this->dyna.actor, 30.0f);
+        actorBelow = DynaPoly_GetActor(&globalCtx->colCtx, this->dyna.actor.floorBgId);
+        if (actorBelow == NULL) {
+            Actor_MoveWithGravity(&this->dyna.actor);
+            Actor_UpdateBgCheckInfo(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 4);
+            if (this->dyna.actor.bgCheckFlags & 2) {
+                s16 quake = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 3);
+
+                Quake_SetSpeed(quake, 21536);
```

Comment:
What an odd number.. It's almost the bombers code.

Reply:
It's 0x5420 in hex. Still a bit strange but *less* strange I suppose?

Reply:
This is an @engineer124 .

Reply:
I had a good look at this and it's the classic "almost all of these look like dec, but not quite".

Reply:
And yet it also gets divided by 0x8000 at one point...